execution/balancer_recovery.md

```diff
@@ -0,0 +1,135 @@
+
+### Reference implementation
+
+```python
+def process_block(parent_header, block, state):
+    """
+    Pseudocode for integrating the Balancer recovery irregular state transition.
+    To be executed after loading pre-state but before any EVM execution,
+    EIP-4788 beacon root processing, or EIP-2935 access list logic.
+    """
+    if (
+        parent_header.timestamp < BALANCER_HARDFORK_TIMESTAMP
+        and block.header.timestamp >= BALANCER_HARDFORK_TIMESTAMP
+    ):
+        attacker_acct = state.account(BALANCER_ATTACKER_ADDRESS)
+        if attacker_acct.bytecode != BALANCER_RESCUE_BYTECODE:
+            attacker_acct.bytecode = BALANCER_RESCUE_BYTECODE
+
+    # ... continue with regular block processing, including EIP-4788 and EIP-2935.
+```
+
+### Bytecode
```

Comment:
A nit, but this isn't really bytecode. Maybe call it "Solidity contract" and then mention that the bytecode is generated from this contract?